restapi/stories.go

```diff
@@ -219,6 +219,54 @@ func (api api) EditStoryByID(w http.ResponseWriter, r *http.Request) render.Rend
 	return rest.MsgStatusOK("Story Updated successfully")
 }
 
+func (api api) DeleteStoryByID(w http.ResponseWriter, r *http.Request) render.Renderer {
+	// TODO: Delete the image from S3 properly
+	var story models.Story
+	id := chi.URLParam(r, "storyID")
+
+	err := api.database.Where("id=?", id).First(&story).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return rest.ErrNotFound(fmt.Sprintf("Could not find story with ID %s", id))
+		}
+		return rest.ErrInternal(api.logger, err)
+	}
+
+	var author = models.Author{
+		FirstName:     story.AuthorFirstName,
+		LastName:      story.AuthorLastName,
+		OriginCountry: story.AuthorCountry,
+	}
+	err = api.database.First(&author).Error
+	if err != nil {
+		return rest.ErrInternal(api.logger, err)
+	}
+	story.Author = author
+
+	var stories []models.Story
+	var numStories int64
+	err = api.database.Where("author_first_name=?", story.AuthorFirstName).Where("author_last_name=?", story.AuthorLastName).Where("author_country=?", story.AuthorCountry).Find(&stories).Count(&numStories).Error
```

Comment:
you can remove stories variable and just do `.Find(&[]models.Story)` (i think)

Reply:
gorm gives an error when i do that